subprojects/soak/src/integTest/groovy/org/gradle/vfs/VirtualFileSystemRetentionSoakTest.groovy

```diff
@@ -124,16 +122,16 @@ class VirtualFileSystemRetentionSoakTest extends DaemonIntegrationSpec implement
             waitBetweenChangesToAvoidOverflow()
         }
         boolean overflowDetected = detectOverflow(daemon, 0)
-        int expectedNumberOfRetainedFiles = overflowDetected
-            ? 0
-            : retainedFilesInCurrentBuild - numberOfChangedSourcesFilesPerBatch
+        int expectedNumberOfRetainedFiles = retainedFilesInCurrentBuild - numberOfChangedSourcesFilesPerBatch
```

Comment:
We should only calculate this if `!overflowDetected`.

Reply:
Done.